### PR TITLE
Fix several build issues in CI when using Clang

### PR DESCRIPTION
**Pull request by czentgr**
A number of issues came up when building the CI pipeline with Clang.

1. When CUDA nvcc uses Clang it complains that totalBytes is set but unused.  The intent seems to have been that the parameter is an in- and output parameter.
2. If benchmarks are not turned on then the CUDA build fails as dependencies are not available.
3. HDFS virtual function definition not explicitly overridden causing an error  (error: 'rename' overrides a member function but is not marked 'override’).

### CODE REVIEW

**Comment by czentgr** [comment]
@majetideepak FYI.

File: velox/experimental/wave/exec/tests/CMakeLists.txt [hunk 1]
```diff
@@ -59,46 +59,48 @@ target_link_libraries(
 
 add_test(velox_wave_exec_test velox_wave_exec_test)
 
-add_executable(velox_wave_benchmark WaveBenchmark.cpp)
+if(${VELOX_ENABLE_BENCHMARKS} OR ${VELOX_ENABLE_BENCHMARKS_BASIC})
```
Comment (majetideepak) [inline velox/experimental/wave/exec/tests/CMakeLists.txt:62]: nit: I think `VELOX_ENABLE_BENCHMARKS` is sufficient here.

Comment (czentgr) [reply velox/experimental/wave/exec/tests/CMakeLists.txt:62]: I suppose it isn't a basic benchmark and should only be enabled along the top level benchmarks - the dependency is available then as well. But perhaps we can get a comment about it.

**Comment by assignUser** [comment]
I know that CUDA/libcudf etc. a much better supported with GCC so potentially switching those parts of when building with clang might be required if hard to solve problems show up.

**Comment by czentgr** [comment]
Yes. We can revisit if problems show up. In this case I believe this was a correct error shown. I read the nvcc also uses a modified LLVM and is similar to Clang actually. So I hope this would work out without major problems going forward.

**Review by kgpai — APPROVED** [review]

**Pull request closed without merging**